sage_modabvar/abvar.py

```diff
@@ -2129,6 +2130,38 @@ def _ambient_hecke_matrix_on_modular_symbols(self, n):
         self.__ambient_hecke_matrix_on_modular_symbols[n] = T
         return T
 
+    def number_of_rational_points(self):
+        """
+        Return the number of rational points of this modular abelian variety.
+
+        It is not always possible to compute the order of the torsion
+        subgroup. The BSD conjecture is assumed to compute the algebraic rank.
+
+        OUTPUT: a positive integer or infinity
+
+        EXAMPLES::
+
+        sage: from sage_modabvar import J0, J1
+        sage: J0(23).number_of_rational_points()
+        11
+        sage: J0(29).number_of_rational_points()
+        7
+        sage: J0(37).number_of_rational_points()
+        +Infinity
+        """
+
+        # The rank of a simple is positive if the lratio is zero.
```

Comment:
make a note that `simple.lseries().lratio()` is potentially a **lot** harder to compute than just whether or not `L(1)=0`.